### PR TITLE
openstack-ardana/crowbar: enable ci.suse.de deployments (SCRD-9264)

### PR DESCRIPTION
**Pull request by stefannica**
Enable cloud.suse.de as an alternative OpenStack platform for
virtual Cloud deployments. This change consists of the following
parts:

 - the `os_cloud` parameter, used in Jenkins jobs and ansible
 playbooks, was previously used to indicate the OpenStack client
 configuration used to connect to the ECP. One could give it different
 values to target different OpenStack ECP projects (e.g the
 `engcloud-cloud-ci` value was used to target the `cloud` project
 used for custom deployments, and `engcloud-cloud-ci-private` to
 target the `cloud-ci` project used exclusively by the CI).
 This parameter is now only used to indicate which OpenStack cloud
 platforms to use to host the virtual environment. It can only take
 two values: `engcloud` and `susecloud`.
 Switching between various OpenStack projects is now accomplished
 with a different parameter, `os_project_name`, which can simply be
 set to the project name value (e.g. `cloud`, `support`).

 - the `cloud-ci` OpenStack project is now reserved and used exclusively
 and automatically in all Jenkins jobs that run with Lockable Resource
 slots. This means that we no longer need to set `os_project_name`
 explicitly to `cloud-ci` when deploying environments associated with the
 Lockable Resource slots.

 - the Lockable Resource slot management is changed to better reflect
 the new situation (2 Cloud products and 2 target Cloud platforms):
   - all Lockable Resource slots associated with virtual cloud
   deployments are renamed to be of the form `<os_cloud>-ci-slot<number>`
   (e.g. `susecloud-ci-slot01`, `engcloud-ci-slot02`), where the
   `os_cloud` particle has the same function as the `os_cloud`
   parameter used in the Jenkins jobs and ansible playbooks:
   it uniquely identifies the target OpenStack Cloud platform.
   - the `cloud-ardana-ci-slot` label will continue to be used
   to tag slots that can accommodate Ardana CI deployments
   - a new `cloud-crowbar-ci-slot` label is introduced that is
   used to tag slots that can accommodate Crowbar CI deployments
   - a new `engcloud-ci-slot` label is introduced that is
   used to tag slots that are used exclusively with the engineering
   cloud platform
   - a new `susecloud-ci-slot` label is introduced that is
   used to tag slots that are used exclusively with the SUSE
   cloud platform
   - two new Lockable Resource slots have been added, one for
   each OpenStack platform: `engcloud-API` and `susecloud-API`.
   They will replace the previous `cloud-ECP-API` slot and will
   be used to independently serialize the calls that the CI
   Jenkins jobs make to the OpenStack API of the two cloud platforms.

 - one periodic cloud image update job per product type (crowbar/ardana)
 and per target cloud platform (engcloud/susecloud) is introduced.
 This results in 10 jobs being created that replace the 5 cloud
 image update jobs that are currently active.
 There is one notable difference: since the SUSE cloud platform
 is currently running Newton, it still doesn't fully support shared
 images, so the visibility of images updated there by the periodic
 CI jobs needs to be `private`.
 
 - the `openstack-ardana-heat` Jenkins job was split into two jobs, each
 one running exclusively for one of the available OpenStack cloud platforms.
 Having one job running exclusively for a target OpenStack platform
 means it can be used to monitor the platforms individually for history
 of failures and overall reliability.

Documentation covering this feature is submitted separately as #3426.

### CODE REVIEW

File: jenkins/ci.suse.de/templates/cloud-heat-template.yaml [hunk 1]
```diff
@@ -100,10 +115,10 @@
     pipeline-scm:
       scm:
         - git:
-            url: ${git_automation_repo}
+            url: ${{git_automation_repo}}
```
Comment (skazi0) [inline jenkins/ci.suse.de/templates/cloud-heat-template.yaml]: was this incorrect before?

Comment (stefannica) [reply jenkins/ci.suse.de/templates/cloud-heat-template.yaml]: @skazi0 yeah, what basically happened is when you convert a JJB job into a template, you have to double all your `{`s  and `}`s, otherwise jenkins-job-builder will think you're trying to expand a template parameter

**Comment by stefannica** [comment]
I tested this with manual builds, as well as Jenkins builds for both Crowbar and Ardana in the ECP and the SUSE cloud. I wasn't able to successfully deploy Crowbar in the SUSE cloud mainly because there were some pending issues with Neutron in the SUSE cloud that prevented me from doing that. This works as expected though with the ECP.

**Comment by skazi0** [comment]
@stefannica rebase

**Comment by stefannica** [comment]
rebased...

File: scripts/jenkins/ardana/manual/deploy-crowbar.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ source lib.sh
 
 validate_input
 setup_ansible_venv
-mitogen_disable
+mitogen_enable
```
Comment (JanZerebecki) [inline scripts/jenkins/ardana/manual/deploy-crowbar.sh]: Why is this here?
Recent discussions sounded like it has problems. Do you know of any? Maybe this warrants its own PR.
This commit has more code changes that I wouldn't have expected. The commit message sounds like it is only supposed to change documentation. Were those intentional?

**Comment by stefannica** [comment]
Rebased

File: scripts/jenkins/ardana/manual/deploy-crowbar.sh (hunk 2 above)
Comment (stefannica) [reply scripts/jenkins/ardana/manual/deploy-crowbar.sh]: This change got merged in the main branch, so it's not in this PR anymore.

File: scripts/jenkins/ardana/ansible/roles/heat_stack/defaults/main.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 heat_action: "create"
 
-heat_stack_create_timeout: 480
+heat_stack_create_timeout: 900
```
Comment (flaviodsr) [inline scripts/jenkins/ardana/ansible/roles/heat_stack/defaults/main.yml]: same here

File: jenkins/ci.suse.de/pipelines/openstack-cloud-heat.Jenkinsfile [hunk 4]
```diff
@@ -67,8 +70,8 @@ pipeline {
           // Dump the heat_template multi-string parameter value into a file
           writeFile file: "$WORKSPACE/heat_template.yml", text: params.heat_template
 
-          lock(resource: 'cloud-ECP-API') {
-            timeout(time: 10, unit: 'MINUTES', activity: true) {
+          lock(resource: "$os_cloud-API") {
+            timeout(time: 20, unit: 'MINUTES', activity: true) {
```
[the remark this answers is not in the document]
Comment (stefannica) [reply jenkins/ci.suse.de/pipelines/openstack-cloud-heat.Jenkinsfile]: @flaviodsr because we use the same Jenkinsfile for both ECP and SUSE cloud platforms, and the SUSE cloud platform is slower (based on Newton, whereas ECP is Pike).

File: scripts/jenkins/ardana/ansible/roles/heat_stack/defaults/main.yml (hunk 3 above)
Comment (stefannica) [reply scripts/jenkins/ardana/ansible/roles/heat_stack/defaults/main.yml]: check this job history to get an idea why: https://ci.suse.de/job/openstack-cloud-heat-susecloud/
